core/jobs/batch_jobs/collection_info_jobs.py

```diff
@@ -0,0 +1,114 @@
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Jobs that extract Collection models information."""
+
+from __future__ import annotations
+
+from core.jobs import base_jobs
+from core.jobs.io import ndb_io
+from core.jobs.transforms import job_result_transforms
+from core.jobs.types import job_run_result
+from core.platform import models
+
+import apache_beam as beam
+
+MYPY = False
+if MYPY: # pragma: no cover
+    from mypy_imports import collection_models
+    from mypy_imports import feedback_models
+    from mypy_imports import user_models
+
+(
+    collection_models, feedback_models, user_models
+) = models.Registry.import_models([
+    models.NAMES.collection, models.NAMES.feedback, models.NAMES.user])
+
+datastore_services = models.Registry.import_datastore_services()
+
+
+class GetCollectionOwnersEmailsJob(base_jobs.JobBase):
+    """Job that generate emails of CollectionModels users."""
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+        """Returns a PCollection of 'SUCCESS' or 'FAILURE' results from
+        generating email of CollectionModel user.
+
+        Returns:
+            PCollection. A PCollection of 'SUCCESS' or 'FAILURE' results from
+            generating email of CollectionModel user.
+        """
+        collection_user_ids_pcollection = (
+            self.pipeline
+            | 'Get all collection rights models' >> ndb_io.GetModels(
+                collection_models.CollectionRightsModel.get_all(
+                    include_deleted=False))
+            | 'Extract user IDs' >> beam.FlatMap(
+                    lambda collection_rights: collection_rights.owner_ids)
+            | 'Remove duplicates' >> beam.Distinct() # pylint: disable=no-value-for-parameter
+        )
+        collection_user_ids = beam.pvalue.AsIter(
+            collection_user_ids_pcollection)
+
+        user_email_pcollection = (
+            self.pipeline
+            | 'Get all user settings models' >> ndb_io.GetModels(
+                user_models.UserSettingsModel.get_all(
+                    include_deleted=False))
+            | 'Filter model that belong to collection' >> (
+                beam.Filter(
+                    lambda model, ids: (
+                        model.id in ids),
+                    ids=collection_user_ids
+                ))
+            | 'Get e-mail' >> beam.Map(lambda model: model.email)
+        )
+
+        return (
+            user_email_pcollection
+            | 'Count the output' >> (
+                job_result_transforms.CountObjectsToJobRunResult())
```

Comment:
This will not return the emails, also we probably need to return both the email and the collection ID that they own.

Reply:
@vojtechjelinek do we need the collection_id ? because to extract it along with the right email makes the job so much more longer

Reply:
Done